tsconfig.json

```
@@ -12,6 +12,6 @@
         "skipLibCheck": true,
         "declaration": true,
         "declarationDir": "./",
```

Jetzt ist die `index.d.ts` ja weg, aber hier wird das wieder dorthin gebaut. Ich würde das wie folgt ändern, um die Typen neben dem Javascript zu haben:

```suggestion
        "declarationDir": "./dist",
```

In der `package.json` dann auch noch Z. 6ff entsprechend mit anpassen:

```json
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist/**/*"
],
```

Dann liegt da alles beisammen

---

Dann wird das aber im package nicht mehr erkannt. Die index.d.ts muss im Root liegen wenn ich nicht irre.

---

Dafür ja das Feld `types` in der package.json. Das ist aktuell: `types: "index.d.ts"`, wenn man das auch (entsprechend der Konfiguration für Typescript) auf `types: "dist/index.d.ts"` ändert, werden die Typen auch in referenzierenden Packages entsprechend gefunden.

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package